api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/VisitsQueryBuilder.java

```diff
@@ -0,0 +1,83 @@
+package org.pmiops.workbench.cohortbuilder.querybuilder;
+
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import org.pmiops.workbench.model.SearchParameter;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * VisitsQueryBuilder is an object that builds {@link QueryJobConfiguration}
+ * for BigQuery for visit criteria types.
+ */
+@Service
+public class VisitsQueryBuilder extends AbstractQueryBuilder {
+
+  private static final String VISIT_SELECT_CLAUSE_TEMPLATE =
+    "select person_id \n" +
+      "from `${projectId}.${dataSetId}.visit_occurrence` a\n";
+
+  private static final String VISIT_CHILD_CLAUSE_TEMPLATE =
+    "where a.visit_concept_id in unnest(${visitConceptIds})\n";
+
+  private static final String VISIT_PARENT_CLAUSE_TEMPLATE =
+    "where a.visit_concept_id in (\n" +
+      "select descendant_concept_id\n" +
+      "from `${projectId}.${dataSetId}.concept_ancestor` \n" +
+      "where ancestor_concept_id in unnest(${parentIds}))\n";
+
+  private static final String UNION_TEMPLATE = " union all\n";
+
+  @Override
+  public QueryJobConfiguration buildQueryJobConfig(QueryParameters inputParameters) {
+    List<String> queryParts = new ArrayList<>();
+    Map<String, QueryParameterValue> queryParams = new HashMap<>();
+
+    List<Long> parentList = new ArrayList<>();
+    List<Long> childList = new ArrayList<>();
+    for (SearchParameter parameter : inputParameters.getParameters()) {
+      if (parameter.getGroup()) {
+        parentList.add(parameter.getConceptId());
+      } else {
+        childList.add(parameter.getConceptId());
```

Comment:
Should we have validation of some sort to check that conceptId is not null? And maybe throw a BadRequestException? If so, then maybe we should add a test case as well?

Reply:
We should be consistent, sure. done.